src/promptflow-evals/promptflow/evals/evaluators/_fluency/_fluency.py

```diff
@@ -64,20 +88,7 @@ def __call__(self, *, question: str, answer: str, **kwargs):
         :return: The fluency score.
         :rtype: dict
         """
-        # Validate input parameters
-        question = str(question or "")
-        answer = str(answer or "")
+        return async_run_allowing_running_loop(self._async_evaluator, question=question, answer=answer, **kwargs)
 
-        if not (question.strip() and answer.strip()):
-            raise ValueError("Both 'question' and 'answer' must be non-empty strings.")
-
-        # Run the evaluation flow
-        llm_output = self._flow(question=question, answer=answer)
-
-        score = np.nan
-        if llm_output:
-            match = re.search(r"\d", llm_output)
-            if match:
-                score = float(match.group())
-
-        return {"gpt_fluency": float(score)}
+    def _to_async(self):
```

Comment:
Can we have __acall__ method and do the async logic here instead of creating a asycn evalutor clas ?

Reply:
I tried that option before. Currently, for flex flow, the flow callable must be either __call__ for class-based flow or a global function for function-based flow. Class methods as flow are not supported today and whether they should be supported requires a lot of discussion. 

Having a separate async class appears to be the cleanest solution for now. It also makes it easier to expose async evaluators in the future when needed.